common-npm-packages/packaging-common/locationUtilities.ts

```diff
@@ -167,6 +167,27 @@ export async function retryOnExceptionHelper<T>(action: () => Promise<T>, maxTri
     }
 }
 
+export async function retryOnNullOrExceptionHelper<T>(action: () => Promise<T>, maxTries: number, retryIntervalInMilliseconds: number): Promise<T> {
+    while (true) {
+        try {
+            var response = await action();
+            if(!response) {
+                throw new Error("Response was null or undefined");
+            }
+            return response;
+        }
+        catch(error) {
+            maxTries--;
+            if (maxTries < 1) {
+                throw error;
+            }
+            tl.debug(`Response was null or undefined. Number of retries left: ${maxTries}`);
+            if (error) { logError(error, LogType.warning); }
```

Comment:
Since this will print out each time something fails, if we are retrying a lot it will generate a lot of warning messages in the pipeline. Can we make this to a lower log level, and the final failure one to log a warning (see line 182)?

Reply:
The only thing worh mentioning here is that if we change the log levels, then the logging behavior deviates from `retryOnExceptionHelper`, which is where this implementation came from.  I would be reluctant to change the logging behavior in `retryOnExceptionHelper`, but can change the logging here if this isn't a concern.

Reply:
I think changing it just here might be okay, and we should update the other retry option to do the same. Retrying a request should be invisible to the user since we are expecting the retry to succeed. The warning message here will pop up on people's pipelines even if the call is eventually successful, and people get irate when they see their prod pipelines littered with warnings

Reply:
For people who want to see if we retried, they can add the system.debug var and see that we are actually retrying before failing